### PR TITLE
Replaces Minio refs in docs with MinIO and links

### PR DESCRIPTION
**Pull request by ebozduman**
Official company name `Minio` became `MinIO`. 
This change required our documentation to be modified for all references to the old name, `Minio` as `MinIO` and the links `....://xxx.minio.io` in the documents as `....://xxx.min.io`

## Description
Here are the changes done in all `../docs/*.md` files.

Used Visual Studio Code Editor regular expressions to find and replace those patterns that match the desired text. Here are the two regular expressions used to complete the job in 2 steps:

1. Set `files to include` to `*.md`, choose `Match Case` and `Use Regular Expression` and replace the links using the following regular expressions:

| Find | Replace |
| ------ | ----------- |
| (//(slack\|docs)\\.\|//)minio\\.io | $1min.io |

The above regular expression, as can be seen, only replaced the links that start with either `slack` or `docs` or nothing before `minio.io`. That excluded `dl.minio.io` and `blog.minio.io` since their `min.io` url versions have not been setup yet.

2. 
| Find | Replace |
| ------ | ----------- |
| \bMinio\b | MinIO |

## Motivation and Context
Introduction of a new company name

## Regression
No

## How Has This Been Tested
Code review

## Types of changes
Company name change in documentation is not a breaking change except the links. However, the links that has been touched are verified and tested.

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#29690142-8925-4afc-9e9c-aa22211625a4) |
<a name="29690142-8925-4afc-9e9c-aa22211625a4"></a>
### 7478-1cf9051/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31236
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ee7f84b76d7c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-iapip9c9ed0e64",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3061,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31236/aws-sdk-go-test-iapip9c9ed0e64: dial tcp 72.28.97.57:31236: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

File: docs/bucket/notifications/README.md [hunk 1]
```diff
@@ -392,7 +392,7 @@ $ curl  "http://localhost:9200/minio_events/_search?pretty=true"
               "source" : {
                 "host" : "127.0.0.1",
                 "port" : "38062",
-                "userAgent" : "Minio (linux; amd64) minio-go/2.0.3 mc/2017-02-15T17:57:25Z"
+                "userAgent" : "MinIO (linux; amd64) minio-go/2.0.3 mc/2017-02-15T17:57:25Z"
```
Comment (kannappanr) [inline docs/bucket/notifications/README.md:395]: This change also needs to be done in `minio-go`.

**Review by kannappanr — DISMISSED** [review]
We can maybe keep the change to `MinIO ...` in user agent and make the corresponding changes in other MinIO SDKs

**Review by donatello — DISMISSED** [review]

**Review by kannappanr — DISMISSED** [review]
LGTM

File: docs/bucket/notifications/README.md (hunk 1 above)
Comment (ebozduman) [reply docs/bucket/notifications/README.md:395]: Good reminder.
Will do when I work on minio-go.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#c5efe1b6-5c8c-4cf0-a573-7d82f41b5955) |
| mint-compression-dist-xl.sh | :x: [more...](#f515b6b3-6a8a-42b4-90e4-f83caff7b688) |
| mint-dist-xl.sh | :x: [more...](#5913caeb-e7aa-4388-bad1-051c30762a38) |
<a name="f515b6b3-6a8a-42b4-90e4-f83caff7b688"></a>
### 7478-6a2251b/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32011
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a3e96409db68:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-2xysrvgf1bys0o",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3062,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32011/aws-sdk-go-test-2xysrvgf1bys0o: dial tcp 72.28.97.57:32011: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="c5efe1b6-5c8c-4cf0-a573-7d82f41b5955"></a>
### 7478-6a2251b/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:32729
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7d84f95361bd:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 50 seconds
(4/14) Running aws-sdk-ruby tests ... done in 13 seconds
(5/14) Running awscli tests ... done in 2 minutes and 1 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... FAILED in 1 minutes and 24 seconds
{
  "name": "mc",
  "duration": "83022",
  "function": "test_put_object_with_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-31107 --quiet --no-color cp --storage-class REDUCED_REDUNDANCY /mint/data/datafile-1-MB myminio/mc-test-bucket-13056/mc-test-object-23742  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-13056/mc-test-object-23742`\nmc: <ERROR> Failed to copy `/mint/data/datafile-1-MB`. Multiple disks failures, unable to write data.\nmc: <ERROR> Session safely terminated. To resume session `mc session resume bGHluPbt`"
}

Executed 6 out of 14 tests successfully.
```
<a name="5913caeb-e7aa-4388-bad1-051c30762a38"></a>
### 7478-6a2251b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:30640
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a243cbf15663:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 1 minutes and 49 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 31 seconds
(8/14) Running minio-dotnet tests ... done in 36 seconds
(9/14) Running minio-go tests ... done in 41 seconds
(10/14) Running minio-java tests ... FAILED in 44 seconds
{
  "name": "minio-java",
  "function": "listIncompleteUploads(String bucketName)",
  "duration": 245,
  "status": "FAIL",
  "error": "java.net.NoRouteToHostException: No route to host (Host unreachable) >>> [java.net.PlainSocketImpl.socketConnect(Native Method), java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350), java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206), java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188), java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392), java.net.Socket.connect(Socket.java:589), okhttp3.internal.platform.Platform.connectSocket(Platform.java:130), okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247), okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167), okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:257), okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135), okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264), okhttp3.RealCall.execute(RealCall.java:93), io.minio.MinioClient.executeReq(MinioClient.java:1097), io.minio.MinioClient.execute(MinioClient.java:1056), io.minio.MinioClient.executeDelete(MinioClient.java:1333), io.minio.MinioClient.abortMultipartUpload(MinioClient.java:4899), io.minio.MinioClient.putObject(MinioClient.java:4087), io.minio.MinioClient.putObject(MinioClient.java:3570), FunctionalTest.listIncompleteUploads_test1(FunctionalTest.java:1604), FunctionalTest.runTests(FunctionalTest.java:2991), FunctionalTest.main(FunctionalTest.java:3121)]"
}

Executed 9 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#6bbd3e43-6f69-4e11-bf3c-68e058770fbb) |
<a name="6bbd3e43-6f69-4e11-bf3c-68e058770fbb"></a>
### 7478-528d61b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:30893
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3454c3f8cdf6:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 5 minutes and 6 seconds
(4/14) Running aws-sdk-ruby tests ... FAILED in 1 minutes and 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "presignedPost(bucket_name,file_name,expires_in_sec,max_byte_size)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-1351cbc81c7f",
    "file_name": "datafile-1-kB",
    "expires_in_sec": 60,
    "max_byte_size": 3145728
  },
  "duration": 60023.99,
  "error": "execution expired",
  "status": "FAIL"
}

Executed 3 out of 14 tests successfully.
```

File: pkg/net/url_test.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func TestURLString(t *testing.T) {
 		{URL{}, ""},
 		{URL{Scheme: "http", Host: "play"}, "http://play"},
 		{URL{Scheme: "https", Host: "play:443"}, "https://play"},
-		{URL{Scheme: "https", Host: "play.minio.io:80"}, "https://play.minio.io:80"},
+		{URL{Scheme: "https", Host: "play.minio.io:80"}, "https://play.min.io:80"},
```
Comment (kannappanr) [inline pkg/net/url_test.go:51]: Needs to change in both places

File: pkg/net/url_test.go [hunk 3]
```diff
@@ -73,7 +73,7 @@ func TestURLMarshalJSON(t *testing.T) {
 	}{
 		{URL{}, []byte(`""`), false},
 		{URL{Scheme: "http", Host: "play"}, []byte(`"http://play"`), false},
-		{URL{Scheme: "https", Host: "play.minio.io:0"}, []byte(`"https://play.minio.io:0"`), false},
+		{URL{Scheme: "https", Host: "play.minio.io:0"}, []byte(`"https://play.min.io:0"`), false},
```
Comment (kannappanr) [inline pkg/net/url_test.go:76]: needs to change in both places

File: pkg/net/url_test.go [hunk 4]
```diff
@@ -104,7 +104,7 @@ func TestURLUnmarshalJSON(t *testing.T) {
 	}{
 		{[]byte(`""`), &URL{}, false},
 		{[]byte(`"http://play"`), &URL{Scheme: "http", Host: "play"}, false},
-		{[]byte(`"https://play.minio.io:0"`), &URL{Scheme: "https", Host: "play.minio.io:0"}, false},
+		{[]byte(`"https://play.min.io:0"`), &URL{Scheme: "https", Host: "play.minio.io:0"}, false},
```
Comment (kannappanr) [inline pkg/net/url_test.go:107]: Needs to change in both places

File: pkg/net/url_test.go [hunk 5]
```diff
@@ -139,7 +139,7 @@ func TestParseURL(t *testing.T) {
 		expectErr   bool
 	}{
 		{"http://play", &URL{Scheme: "http", Host: "play"}, false},
-		{"https://play.minio.io:0", &URL{Scheme: "https", Host: "play.minio.io:0"}, false},
+		{"https://play.min.io:0", &URL{Scheme: "https", Host: "play.minio.io:0"}, false},
```
Comment (kannappanr) [inline pkg/net/url_test.go:142]: needs to change in both places

**Comment by kannappanr** [comment]
Please fix travis failures. It looks like there are few places where you still  have `play.minio.io`

**Comment by kannappanr** [comment]
opened #7494 with fixes to unit tests. Commits from this PR are being used in that PR. Closing this PR now.

**Pull request closed without merging**